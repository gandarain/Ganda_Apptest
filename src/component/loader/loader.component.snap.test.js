import React from 'react'
import renderer from 'react-test-renderer'

import Loader from './loader.component'

describe('Loader Component', () => {
  const mockProps = {
    visible: true,
    onClose: jest.fn()
  }

  it('should render loader when visible is true', () => {
    const rendered = renderer.create(<Loader {...mockProps} />)
    const tree = rendered.toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('should render loader when visible is false', () => {
    mockProps.visible = false
    const rendered = renderer.create(<Loader {...mockProps} />)
    const tree = rendered.toJSON()

    expect(tree).toMatchSnapshot()
  })
})
