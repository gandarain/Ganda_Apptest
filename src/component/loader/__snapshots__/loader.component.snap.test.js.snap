// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader Component should render loader when visible is false 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  onRequestClose={[MockFunction]}
  transparent={true}
  visible={false}
/>
`;

exports[`Loader Component should render loader when visible is true 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  onRequestClose={[MockFunction]}
  transparent={true}
  visible={true}
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "rgba(0, 0, 0, 0.7)",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <ActivityIndicator
      color="#FF6049"
      size="large"
    />
  </View>
</Modal>
`;
